blazor/datagrid/column-menu.md

```diff
@@ -90,45 +122,284 @@ In the following sample, Order ID, Order Date, and Freight columns are sorted at
     </GridSortSettings>
     <GridEvents ColumnMenuItemClicked="ColumnMenuItemClickedHandler" TValue="Order"></GridEvents>
     <GridColumns>
-        <GridColumn Field=@nameof(Order.OrderID)  HeaderText="Order ID" TextAlign="TextAlign.Center" Width="120"></GridColumn>
-        <GridColumn Field=@nameof(Order.CustomerID) HeaderText="Customer Name" Width="150" TextAlign="TextAlign.Center" ShowColumnMenu="false"></GridColumn>
-        <GridColumn Field=@nameof(Order.OrderDate) HeaderText="Order Date" Format="d" Type="ColumnType.Date" TextAlign="TextAlign.Center" Width="130"></GridColumn>
-        <GridColumn Field=@nameof(Order.Freight) HeaderText="Freight" Format="C2" TextAlign="TextAlign.Center" Width="120"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.OrderID) HeaderText="Order ID" TextAlign="TextAlign.Center" Width="120"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.CustomerID) HeaderText="Customer Name" Width="150" TextAlign="TextAlign.Center" ShowColumnMenu="false"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.OrderDate) HeaderText="Order Date" Format="d" Type="ColumnType.Date" TextAlign="TextAlign.Center" Width="130"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.Freight) HeaderText="Freight" Format="C2" TextAlign="TextAlign.Center" Width="120"></GridColumn>
     </GridColumns>
 </SfGrid>
+@code {
+    private SfGrid<OrderData> Grid;
+    public List<string> columns = new List<string>() { "OrderID" };
 
-@code{
-    SfGrid<Order> Grid { get; set; }
-    public List<Order> Orders { get; set; }
-    public List<string> columns = new List<string>(){ "OrderID"};
-
+    public List<OrderData> Orders { get; set; }
+   
+    protected override void OnInitialized()
+    {
+        Orders = OrderData.GetAllRecords();
+    }
     public void ColumnMenuItemClickedHandler(ColumnMenuClickEventArgs args)
     {
-     Grid.ClearSortingAsync(columns);
+        Grid.ClearSortingAsync(columns);
```

Comment:
sample is not working properly. check by clearing sorting for more than one column

Reply:
working based on our definition.